### ISSUE TITLE
Can't configure

### ISSUE THREAD

**Issue by ecrona** [ZenProjects/Apache-Authmemcookie-Module#12]
Hey

I tried to install this module, but I can't seem to configure it in Linux.

I've installed libevent-2.1.8, memcached-1.4.36 and libmemcached-1.0.18

And then when I run ...
`./configure --with-libmemcached=/usr/local/include/libmemcached/`

... I get this

> checking for gcc... gcc
> checking for C compiler default output file name... a.out
> checking whether the C compiler works... yes
> checking whether we are cross compiling... no
> checking for suffix of executables... 
> checking for suffix of object files... o
> checking whether we are using the GNU C compiler... yes
> checking whether gcc accepts -g... yes
> checking for gcc option to accept ISO C89... none needed
> checking how to run the C preprocessor... gcc -E
> checking for grep that handles long lines and -e... /bin/grep
> checking for egrep... /bin/grep -E
> checking for ANSI C header files... yes
> checking for libmemcached library... configure: error: /usr/local/include/libmemcached/ not found.

When i check its contents ...
`ls -la /usr/local/include/libmemcached/`

... I get this

> total 20
> drwxr-xr-x 2 root root 4096 may  4 15:37 .
> drwxr-xr-x 8 root root 4096 may  4 15:37 ..
> -rw-r--r-- 1 root root 1713 may  4 15:37 memcached.h
> -rw-r--r-- 1 root root 1717 may  4 15:37 memcached.hpp
> -rw-r--r-- 1 root root 1714 may  4 15:37 util.h

I've tried with **/usr** as the configure path, and some other paths, but it can't seem to find it.

Thanks for any support

**Comment by mcarbonneaux** [comment]
can you send the config.log file ? you used what version of the authmemcookie module ? on what linux distribution ?

**Comment by ecrona** [comment]
Hey

I solved it, apparently the configure file was searching /usr/local/lib64 folder (or actually it just needed to know it existed) for my Ubuntu 14.04, but the files were located in /usr/local/lib.

I simply created a /usr/local/lib64 folder, and it worked

**Comment by mcarbonneaux** [comment]
in fact --with-libmemcached option MUST point to base install directory of libmemcached...
in your case you use:
```
# ./configure --with-libmemcached=/usr/local/include/libmemcached
```
and you must use:
```
# ./configure --with-libmemcached=/usr/local/
```

Effectively they search "--with-libmemcached" dir + lib or lib64 (is exist and on 64bit system only) in m4/ax_libmemcached.m4:
```
    LONG_BIT=$(getconf LONG_BIT)
    LIBDIR=lib
    if test "$LONG_BIT" -eq 64; then 
      if test -d $with_libmemcached/lib64; then
        LIBDIR=lib64
      fi
    fi
```
and after they search memcached.h file:
```
    LIBMEMCACHED_INCLUDE_DIR=$with_libmemcached/include
    LIBMEMCACHED_LIB_DIR=$with_libmemcached/$LIBDIR
    # make sure that a well known include file exists
    if test -e $LIBMEMCACHED_INCLUDE_DIR/libmemcached-1.0/memcached.h && test -d     $LIBMEMCACHED_LIB_DIR; then
```

**Comment by einfallstoll** [comment]
README is misleading because the example is

    ./configure --with-apxs=/path/to/apache/httpd/bin/apxs --with-libmemcached=/path/to/libmemcached/

instead of some real world example.

It would be nice to have a note that it need the parent directory of the `include` and `lib`/`lib64` directory and that the library must be in a folder called `include/libmemcached-1.0`

**Comment by mcarbonneaux** [comment]
The name include/libmemcached-1.0 is included in all installation of v1.0.x of libmemcached...

and generaly when you specify install directory in configure script you point generaly on the install dir, not the include dir...

**Comment by einfallstoll** [comment]
I installed `libmemcached` from a repository and it didn't include the version, so I had to create a symlink at first.

This might be true and logical for you, but not for me (and others). Such a compilation means a lot of effort and research to me. It's just not my daily business and therefore I lack a lot of knowledge and probably won't ever get enough to reach your level. In general I can help myself out really quick read through a few docs and files and know where my installation break, but it would be really nice if you had included this, especially for others that don't know how to help themselves out yet.

I actually just want to document my issue so others may find it in the future.
